### PR TITLE
Run all JS processing through browserify

### PR DESCRIPTION
**Pull request by lunelson**
This branch contains a slew of small tweaks, mostly related to the removal of the JS processors altogether, as the extensions for defining which files to process are already hard-coded in the helpers. 

I hope it's clear enough and not too messy, 😁  — feel free to just crush all these commits in to one, or whatever

### CODE REVIEW

File: lib/helpers/raw.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+    var harpConfig = exports.harpConfig = {};
+  }
+}
+
+harpConfig = _.assign({
+  debug: true,
+  minify: false
+}, harpConfig)
+
+/**
```
Comment (jimjkelly) [inline lib/helpers/raw.js:26]: I'm not too familiar with lodash - what's the difference between the above and

```
harpConfig = Object.assign({
    debug: true,
    minify: false
}, harpConfig)
```

?

**Comment by jimjkelly** [comment]
So a couple things on this - one, this branch is actually just me messing around trying to get sourcemaps through, the actual branch I was working on that's closest to being ready for a PR is the [babelify-browserify](https://github.com/jimjkelly/terraform/tree/babelify-browserify) branch. Sorry, I know I have a bunch of similarly named branches now.  :/

Also -  there are a lot of good changes in here, but I think we probably have the best chance to get this in the more focused it is, there are a lot of changes in here related to changing other things, are they strictly necessary to remove the processors code for JavaScript?

**Comment by lunelson** [comment]
Hey, no you're right not all of these changes are necessary for the PR, and the branch I'm trying to merg here is also a lot of messing around. I suppose I figured to do a PR against sintaxi's master we'd cherry pick from the results of this merge, to make a new branch with probably just one commit on it that contains the essence, namely to "remove JS processors and run everything through browserify, be it coffeescript or as ES6 javascript" or something to that effect.

Re: that use of lodash, again you're right there's no difference from `Object.assign` there but lodash is already included in terraform (from an earlier PR of mine) and I'm not sure if `Object.assign` is available in all versions of Node (I have only a vague sense of what differs in versions of Node, but I heard Node 6 is the one finally supporting almost all of ES6, and I usually use 4 or 5...).

Do you want me to re-do this PR against your other branch and/or crush the commits?

**Comment by jimjkelly** [comment]
Ah, you are right! I hadn't seen the lodash requirement before.  Makes sense.

And yeah, if you wouldn't mind redoing the PR against the other branch, and also if you could just do the relevant commits that'd probably be helpful - with all the other changes it raises the cognitive load thinking about the PR, and in the off chance the other changes cause an unrelated problem it just increases debug time.  Easier (for me at least) to keep things focused.

**Comment by jimjkelly** [comment]
Also, I'm back in UTC+2, which I believe you are as well, so easier syncing up timezone-wise now. :)

**Comment by lunelson** [comment]
Hey, just FYI I still intend to do this but I'm on vacation with limited access—it might be a little while 😁

**Comment by jimjkelly** [comment]
No worries - it's August, to be expected.  :)

I've been super busy myself.  Looking to revisit this in a week or so.

**Pull request closed without merging**